ide/che-core-ide-stacks/src/main/resources/stacks.json

```diff
@@ -2015,5 +2015,105 @@
       "name": "type-zend.svg",
       "mediaType": "image/svg+xml"
     }
+  },
+  {
+    "id": "platformio",
+    "creator": "ide",
+    "name": "PlatformIO",
+    "description": "Stack for IoT development",
+    "scope": "general",
+    "tags": [
+      "Platformio",
+      "pio",
+      "Arduino"
+    ],
+    "components": [
+      {
+        "name": "Python",
+        "version": "3.5.1"
+      },
+      {
+        "name": "pip",
+        "version": "8.1.1"
+      },
+      {
+        "name": "PlatformIO",
+        "version": "3.2.0"
+      }
+    ],
+    "source": {
+      "type": "image",
+        "origin": "ksmster/pio:latest"
+    },
+    "workspaceConfig": {
+      "environments": {
+        "default": {
+          "machines": {
+            "dev-machine": {
+              "agents": [
+                "org.eclipse.che.terminal", "org.eclipse.che.ws-agent", "org.eclipse.che.ssh"
+              ],
+              "servers": {},
+              "attributes" : {
+                "memoryLimitBytes": "2147483648"
+              }
+            }
+          },
+          "recipe": {
+            "location": "ksmster/pio:latest",
+            "type": "dockerimage"
+          }
+        }
+      },
+      "name": "default",
+      "defaultEnv": "default",
+      "description": null,
+      "commands": [
+        {
+          "name": "run",
+          "type": "custom",
+          "commandLine": "cd ${current.project.path} && pio run",
+          "attributes": {
+            "previewUrl": ""
+          }
+        },
+        {
+          "name": "upload",
+          "type": "custom",
+          "commandLine": "cd ${current.project.path} && pio run -t upload",
+          "attributes": {
+            "previewUrl": ""
+          }
+        },
+        {
+          "name": "clean",
+          "type": "custom",
+          "commandLine": "cd ${current.project.path} && pio run -t clean",
+          "attributes": {
+            "previewUrl": ""
+          }
+        },
+        {
+          "name": "Remote: Device list",
+          "type": "custom",
+          "commandLine": "pio remote device list",
+          "attributes": {
+            "previewUrl": ""
+          }
+        },
+        {
+          "name": "Remote: Upload",
+          "type": "custom",
+          "commandLine": "cd ${current.project.path} && pio remote run -t upload",
+          "attributes": {
+            "previewUrl": ""
+          }
+        }
```

Comment:
Would be good to have here "Remote: Serial Port Monitor" mapped to `pio remote device monitor` command.